out/can_database_can_s.c

```diff
@@ -0,0 +1,257 @@
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/**
+ * This file was generated by cantools version 38.0.2 Fri May 12 13:43:11 2023.
+ */
+
+#include <string.h>
+
+#include "can_database_can_s.h"
+
+static inline uint8_t pack_left_shift_u8(
+    uint8_t value,
+    uint8_t shift,
+    uint8_t mask)
+{
+    return (uint8_t)((uint8_t)(value << shift) & mask);
+}
+
+static inline uint8_t pack_left_shift_u32(
+    uint32_t value,
+    uint8_t shift,
+    uint8_t mask)
+{
+    return (uint8_t)((uint8_t)(value << shift) & mask);
+}
+
+static inline uint8_t pack_right_shift_u32(
+    uint32_t value,
+    uint8_t shift,
+    uint8_t mask)
+{
+    return (uint8_t)((uint8_t)(value >> shift) & mask);
+}
+
+static inline uint32_t unpack_left_shift_u32(
+    uint8_t value,
+    uint8_t shift,
+    uint8_t mask)
+{
+    return (uint32_t)((uint32_t)(value & mask) << shift);
+}
+
+static inline uint8_t unpack_right_shift_u8(
+    uint8_t value,
+    uint8_t shift,
+    uint8_t mask)
+{
+    return (uint8_t)((uint8_t)(value & mask) >> shift);
+}
+
+static inline uint32_t unpack_right_shift_u32(
+    uint8_t value,
+    uint8_t shift,
+    uint8_t mask)
+{
+    return (uint32_t)((uint32_t)(value & mask) >> shift);
+}
+
+int can_database_can_s_oct_gps_2_pack(
+    uint8_t *dst_p,
+    const struct can_database_can_s_oct_gps_2_t *src_p,
+    size_t size)
+{
+    if (size < 8u) {
+        return (-EINVAL);
+    }
+
+    memset(&dst_p[0], 0, 8);
+
+    dst_p[0] |= pack_left_shift_u32(src_p->oct_gps_time_uct, 0u, 0xffu);
+    dst_p[1] |= pack_right_shift_u32(src_p->oct_gps_time_uct, 8u, 0xffu);
+    dst_p[2] |= pack_right_shift_u32(src_p->oct_gps_time_uct, 16u, 0xffu);
+    dst_p[3] |= pack_right_shift_u32(src_p->oct_gps_time_uct, 24u, 0x01u);
+    dst_p[3] |= pack_left_shift_u8(src_p->oct_gps_num_sats, 1u, 0xfeu);
+    dst_p[4] |= pack_left_shift_u8(src_p->oct_gps_hdop, 0u, 0xffu);
+
+    return (8);
+}
+
+int can_database_can_s_oct_gps_2_unpack(
+    struct can_database_can_s_oct_gps_2_t *dst_p,
+    const uint8_t *src_p,
+    size_t size)
+{
+    if (size < 8u) {
+        return (-EINVAL);
+    }
+
+    dst_p->oct_gps_time_uct = unpack_right_shift_u32(src_p[0], 0u, 0xffu);
+    dst_p->oct_gps_time_uct |= unpack_left_shift_u32(src_p[1], 8u, 0xffu);
+    dst_p->oct_gps_time_uct |= unpack_left_shift_u32(src_p[2], 16u, 0xffu);
+    dst_p->oct_gps_time_uct |= unpack_left_shift_u32(src_p[3], 24u, 0x01u);
+    dst_p->oct_gps_num_sats = unpack_right_shift_u8(src_p[3], 1u, 0xfeu);
+    dst_p->oct_gps_hdop = unpack_right_shift_u8(src_p[4], 0u, 0xffu);
+
+    return (0);
+}
+
+uint32_t can_database_can_s_oct_gps_2_oct_gps_time_uct_encode(double value)
```

Comment:
I think these function names are getting a bit out of hand...

I think it would be better to have database names of `canc_db` and `cans_db` rather than `can_database_can_s` etc. Just to improve readability and avoid having "CAN" twice everywhere.

Reply:
So the files would become `can_s.h` and `can_s.c` etc. Then the function names will be slightly more reasonable in length.